packages/@sanity/cli/src/commands/blueprints/addBlueprintsCommand.ts

```diff
@@ -0,0 +1,82 @@
+import {type CliCommandDefinition} from '../../types'
+
+const helpText = `
+Arguments
+  [type]  Type of Resource to add (currently only 'function' is supported)
+
+Examples
+  # Add a Function Resource
+  sanity blueprints add function
+`
+
+const addBlueprintsCommand: CliCommandDefinition = {
+  name: 'add',
+  group: 'blueprints',
+  helpText,
+  signature: '<type>',
+  description: 'Add a Resource to a Blueprint',
+  hideFromHelp: true,
+  async action(args, context) {
+    const {output, prompt} = context
+    const {print} = output
+
+    const [resourceType] = args.argsWithoutOptions
+
+    if (!resourceType) {
+      print('Resource type is required. Available types: function')
+      return
+    }
+
+    const {blueprint: blueprintAction, resources: resourcesAction} = await import(
+      '@sanity/runtime-cli/actions/blueprints'
+    )
+
+    const existingBlueprint = blueprintAction.findBlueprintFile()
+    if (!existingBlueprint) {
+      print('No blueprint file found. Run `sanity blueprints init` first.')
+      return
+    }
+
+    if (resourceType === 'function') {
+      const functionName = await prompt.single({
+        type: 'input',
+        message: 'Enter function name:',
+        validate: (input: string) => input.length > 0 || 'Function name is required',
+      })
+
+      const functionType = await prompt.single({
+        type: 'list',
+        message: 'Choose function type:',
+        choices: [
+          {value: 'document-mutation', name: 'Document Mutation'},
+          {value: 'document-publish', name: 'Document Publish'},
+          {value: 'document-update', name: 'Document Update'},
+          {value: 'document-delete', name: 'Document Delete'},
+        ],
```

Comment:
Should we be giving folks all these options if we only support mutation currently?

Reply:
Good question! a copy-paste from runtime-cli. I don't think we should allow these choices.

 I will update with real types and only enable the mutation.  
Though I fear the current type `"sanity.function.mutation"` found in the Blueprints spec is too generic right? We need a better list of function types.

Reply:
After looking into it more and learning how Functions is working right now, I think the correct default value right now is `sanity.function.document-publish`. I updated this PR to reflect that.